libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -91,3 +94,62 @@ void AP_BattMonitor_Backend::update_resistance_estimate()
     // update estimated voltage without sag
     _state.voltage_resting_estimate = _state.voltage + _state.current_amps * _state.resistance;
 }
+
+bool AP_BattMonitor_Backend::extended_arming_checks(bool report) const
+{
+    char failure_str[41]; // only 50 characters fit in statustext, reserve 9 for "Battery X"
+    float voltage;
+
+    switch (_params.failsafe_voltage_source()) {
+        case AP_BattMonitor_Params::BattMonitor_LowVoltageSource_Raw:
+        default:
+            voltage = _state.voltage;
+            break;
+        case AP_BattMonitor_Params::BattMonitor_LowVoltageSource_SagCompensated:
+            voltage = _mon.voltage_resting_estimate(_instance);
```

Comment:
I understand this, but I would prefer to see the method move to backend and then the frontend one would call the backend - instance would then not be needed.

Reply:
Instance is still needed for sending the status text messages down, although it could be prefixed by AP_BattMonitor directly I guess.